homeassistant/components/matter/cover.py

```diff
@@ -0,0 +1,154 @@
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Matter Cover from Config Entry."""
+    matter = get_matter(hass)
+    matter.register_platform_handler(Platform.COVER, async_add_entities)
+
+
+class MatterCover(MatterEntity, CoverEntity):
+    """Representation of a Matter Cover."""
+
+    entity_description: CoverEntityDescription
+    _attr_supported_features = (
+        CoverEntityFeature.OPEN
+        | CoverEntityFeature.CLOSE
+        | CoverEntityFeature.STOP
+        | CoverEntityFeature.SET_POSITION
+    )
+
+    @property
+    def current_cover_position(self) -> int:
+        """Return the current position of cover."""
+        if self._attr_current_cover_position:
+            current_position = self._attr_current_cover_position
+        else:
+            current_position = self.get_matter_attribute_value(
+                clusters.WindowCovering.Attributes.CurrentPositionLiftPercentage
+            )
+
+        assert current_position is not None
+
+        return current_position
+
+    @property
+    def is_closed(self) -> bool:
+        """Return true if cover is closed, else False."""
+        return self.current_cover_position == 0
+
+    async def async_stop_cover(self, **kwargs: Any) -> None:
+        """Stop the cover movement."""
+        await self.send_device_command(clusters.WindowCovering.Commands.StopMotion())
+
+    async def async_open_cover(self, **kwargs: Any) -> None:
+        """Open the cover."""
+        await self.send_device_command(clusters.WindowCovering.Commands.UpOrOpen())
+
+    async def async_close_cover(self, **kwargs: Any) -> None:
+        """Close the cover."""
+        await self.send_device_command(clusters.WindowCovering.Commands.DownOrClose())
+
+    async def async_set_cover_position(self, **kwargs: Any) -> None:
+        """Set the cover to a specific position."""
+        position = kwargs[ATTR_POSITION]
+        await self.send_device_command(
+            clusters.WindowCovering.Commands.GoToLiftValue(position)
+        )
+
+    async def send_device_command(self, command: Any) -> None:
+        """Send device command."""
+        await self.matter_client.send_device_command(
+            node_id=self._endpoint.node.node_id,
+            endpoint_id=self._endpoint.endpoint_id,
+            command=command,
+        )
+
+    @callback
+    def _update_from_device(self) -> None:
+        """Update from device."""
+        operational_status = self.get_matter_attribute_value(
+            clusters.WindowCovering.Attributes.OperationalStatus
+        )
+
+        assert operational_status is not None
+
+        LOGGER.debug(
+            "Operational status %s for %s",
+            f"{operational_status:#010b}",
+            self.entity_id,
+        )
+
+        state = operational_status & OPERATIONAL_STATUS_MASK
+        match state:
+            case OperationalStatus.covering_is_currently_opening.value:
```

Comment:
the point of the enum is a bit gone now this way.
suggestions:

1) Add a missing method in the enum to fallback to "unknown" key --> read the operational status as enum.
2) drop the enum and just use constants instead
3) switch to IntEnum so you can do the comparing without the need for the ".value"